src/nupic/algorithms/KNNClassifier.py

```diff
@@ -201,6 +207,7 @@ class labels
     self.replaceDuplicates = replaceDuplicates
     self.cellsPerCol = cellsPerCol
     self.maxStoredPatterns = maxStoredPatterns
+    self.minSparsity = minSparsity
```

Comment:
I believe you need to add something in `__getstate__()`/`__setstate__()` to handle deserialized objects for which `self.minSparsity` has not been set.  Otherwise https://github.com/subutai/nupic/blob/d6e04686f78a27e1dbd5ed20ac6c283dba94ccdd/src/nupic/algorithms/KNNClassifier.py#L545 can cause an `AttributeError` exception, as I'm currently experiencing in Imbu.